packages/ddex/processor/src/s3poller.ts

```diff
@@ -4,82 +4,81 @@ import {
   S3Client,
   S3ClientConfig,
 } from '@aws-sdk/client-s3'
-import { fromIni } from '@aws-sdk/credential-provider-ini'
+import { mkdir, readFile, stat, writeFile } from 'fs/promises'
 import { basename, dirname, join, resolve } from 'path'
 import { s3markerRepo } from './db'
 import { parseDdexXml } from './parseDelivery'
-import { mkdir, readFile, stat, writeFile } from 'fs/promises'
-
-export function dialS3() {
-  const { AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY } = process.env
-  if (!AWS_ACCESS_KEY_ID || !AWS_SECRET_ACCESS_KEY) {
-    throw new Error(`AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY required`)
-  }
-
-  const config: S3ClientConfig = {
-    credentials: {
-      accessKeyId: AWS_ACCESS_KEY_ID,
-      secretAccessKey: AWS_SECRET_ACCESS_KEY,
-    },
-    // region: process.env.AWS_REGION
-  }
-  if (process.env.AWS_ENDPOINT) {
-    config.endpoint = process.env.AWS_ENDPOINT
-    config.forcePathStyle = true
+import { SourceConfig, sources } from './sources'
+
+// key by bucket?
+const s3clients: Record<string, S3Client> = {}
```

Comment:
so this is keyed by access key?

type AccessKey = string
could do Record<AccessKey, S3Client> = {}